src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/PublicationServiceTests.cs

```diff
@@ -32,6 +32,452 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
+
+            publicationRepository.Setup(s => s.GetPublicationListForUser(userId, topic.Id))
+                .ReturnsAsync(new List<Publication>
+                {
+                    publication,
+                });
+            publicationRepository.Setup(s => s.IsSuperseded(publication)).Returns(false);
+
+            var userService = new Mock<IUserService>(Strict);
+
+            userService.Setup(s => s.GetUserId()).Returns(userId);
+            userService.Setup(s => s.MatchesPolicy(CanAccessSystem)).ReturnsAsync(true);
+            userService.Setup(s => s.MatchesPolicy(CanViewAllPublications)).ReturnsAsync(false);
+
+            userService.Setup(s => s.MatchesPolicy(publication, CanUpdateSpecificPublication)).ReturnsAsync(true);
+            userService.Setup(s => s.MatchesPolicy(CanUpdatePublicationTitles)).ReturnsAsync(true);
+            userService.Setup(s => s.MatchesPolicy(CanUpdatePublicationSupersededBy)).ReturnsAsync(true);
+            userService.Setup(s => s.MatchesPolicy(publication, CanCreateReleaseForSpecificPublication)).ReturnsAsync(true);
+            userService.Setup(s => s.MatchesPolicy(publication, CanAdoptMethodologyForSpecificPublication)).ReturnsAsync(false);
+            userService.Setup(s => s.MatchesPolicy(publication, CanCreateMethodologyForSpecificPublication)).ReturnsAsync(false);
+            userService.Setup(s => s.MatchesPolicy(publication, CanManageExternalMethodologyForSpecificPublication)).ReturnsAsync(false);
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var publicationService = BuildPublicationService(
+                    context: contentDbContext,
+                    userService: userService.Object,
+                    publicationRepository: publicationRepository.Object);
+
+                var result = await publicationService
+                    .GetPublicationsByTopic(permissions: true, topic.Id);
+
+                VerifyAllMocks(publicationRepository);
+
+                var publicationViewModelList = result.AssertRight();
+
+                var publicationViewModel = Assert.Single(publicationViewModelList);
+
+                Assert.NotNull(publicationViewModel.Permissions);
+                Assert.True(publicationViewModel.Permissions!.CanUpdatePublication);
+                Assert.True(publicationViewModel.Permissions.CanUpdatePublicationTitle);
+                Assert.True(publicationViewModel.Permissions.CanUpdatePublicationSupersededBy);
+                Assert.True(publicationViewModel.Permissions.CanCreateReleases);
+                Assert.False(publicationViewModel.Permissions.CanAdoptMethodologies);
+                Assert.False(publicationViewModel.Permissions.CanCreateMethodologies);
+                Assert.False(publicationViewModel.Permissions.CanManageExternalMethodology);
+            }
+        }
+
```

Comment:
We should test the case where a `topicId` is required if the user can see all publications

Reply:
I think I've covered this, but I'm not 100% so I'll leave this unresolved.